### ISSUE TITLE
Unclear what the latest version is

### ISSUE THREAD

**Issue by ximion** [BlueBrain/mod2c#88]
Hi!
The latest and most current tag and marked release for this project is version `0.9` from Oct 5, 2016, which previously on Aug 30, 2016 version `1.2.0` was tagged.
This is very confusing, especially for Linux distributions which package this tool.
Could it be clarified what the latest version is, either by renaming the 0.9 tag or by renaming the other tag, depending on what the intended version is? (easiest solution and probably the one with the least amount of chaos would likely be to just make a new release with a version number higher than `1.2.0`)

Thank you! :-)

**Comment by pramodk** [comment]
Hello @ximion !

You are completely right! 

We stopped tagging new releases when we decided to use mod2c exclusively as git submodule under [BlueBrain/CoreNeuron](https://github.com/BlueBrain/CoreNeuron) and now [neuronsimulator/nrn](https://github.com/neuronsimulator/nrn). Also, this (older) tool is not a general purpose but only meant to use internally by NEURON/CoreNEURON simulator. And that's when we stopped taking care of releases. (New alternative is https://github.com/BlueBrain/nmodl).

We can make a new release if this is desirable / helpful for distributions. Could you just tell us a bit about where/how you are packaging this? (and based on above comments, do you think this stills need to be distributed independently?).

**Comment by ximion** [comment]
Hi!
Thank you for the reply! The NEURON packaging is in Debian's GitLab instance, but not yet published as the package is currently under policy and license review (I hope to get an exception from the release team to be able to update NEURON to 8.2.x in the next release). => https://salsa.debian.org/science-team/nrn-mod2c

In Debian anything that *could* be used by multiple projects needs to be in a separate package, as having multiple code copies in multiple places is a nightmare for both license review/compliance as well as fixing security issues (if there is a code copy in 300 places, a security issue needs to be addressed 300 times, while if there is one shared code copy, it only needs to be fixed once). It's a matter of keeping a software distribution maintainable for years without the task becoming too overwhelming. (see https://wiki.debian.org/UpstreamGuide#No_inclusion_of_third_party_code )

As for this particular package: If it's only used by NEURON, I think just merging the code into the Neuron repository is likely the easiest thing to do, then you don't even need to worry about separate releases or keeping a submodule up to date anymore.
So, it looks like just merging this with Neuron is an easy way, it would also actually make packaging Neuron a bit easier ;-)

Otherwise at least making a release with a higher version number would be good - we can package Git snapshots of upstream projects, but if there are newer tags, it confuses a bunch of automation on our side that alerts about new upstream versions (which in turn confuses people looking at the messages ^^)

**Comment by pramodk** [comment]
> As for this particular package: If it's only used by NEURON, I think just merging the code into the Neuron repository is likely the easiest thing to do, then you don't even need to worry about separate releases or keeping a submodule up to date anymore.
> So, it looks like just merging this with Neuron is an easy way, it would also actually make packaging Neuron a bit easier ;-)

Completely agree! (in this specific case, we want to remove mod2c usage soon)

> Otherwise at least making a release with a higher version number would be good - we can package Git snapshots of upstream projects, but if there are newer tags, it confuses a bunch of automation on our side that alerts about new upstream versions

Ok. But just to be sure: if there is a newer tag, you want something that is compatible with the latest neuron release like 8.2.x, right?

**Comment by ximion** [comment]
> Ok. But just to be sure: if there is a newer tag, you want something that is compatible with the latest neuron release like 8.2.x, right?

Yes, that would be nice! At the moment we're using a Git snapshot: https://tracker.debian.org/pkg/nrn-mod2c (which works as well).
If the removal of this tool is imminent, having a new release tag may not even be needed.

**Comment by pramodk** [comment]
@ximion : Sorry for the late response here - as the deprecation of this software is still ongoing and will take still few more weeks, I created tag [8.2.2](https://github.com/BlueBrain/mod2c/releases/tag/8.2.2). In case this simplify the release process upstream.
